apps/examples/webserver_test/webserver_test_main.c

```diff
@@ -0,0 +1,1293 @@
+	param->async_flag = WGET_ERR;
+	FUNC_EX;
+	return WGET_ERR;
+}
+#endif
+
+int ws_test_client_send_request(struct http_client_request_t *request)
+{
+	FUNC_EN;
+	if (request == NULL) {
+		PRNT("Error: Request is null");
+		return -1;
+	}
+	if (request->method < WGET_MODE_GET || request->method > WGET_MODE_DELETE) {
+		PRNT("Error: Incorrect method value!!");
+		return -1;
+	}
+	if (request->buffer == NULL) {
+		PRNT("Error: Buffer is NULL!!");
+		return -1;
+	}
+	if (request->buflen <= 0) {
+		PRNT("Error: Buffer length must be bigger than 0!!");
+		return -1;
+	}
+	if (request->encoding != CONTENT_LENGTH && request->encoding != CHUNKED_ENCODING) {
+		PRNT("Error: Incorrect encoding value!!");
+		return -1;
+	}
+
+	// sync call
+	int ret = ws_test_wget_base((void *)request);
+	FUNC_EX;
+	return ret;
+}
+
+static int ws_test_http_client_send_request(struct http_client_request_t *request,
+		 void *ssl_config, struct http_client_response_t *response, wget_callback_t cb)
+{
+	FUNC_EN;
+	int res = -1;
+#ifdef CONFIG_NET_SECURITY_TLS
+	struct mallinfo data;
+	struct http_client_ssl_config_t *ssl_conf = ssl_config;
+#endif
+
+	if (request == NULL) {
+		PRNT("Error: Request is null");
+		goto ret_error;
+	}
+
+	request->tls = false;
+	request->response = response;
+	request->callback = cb;
+
+	if (request->url == NULL) {
+		PRNT("Error: URL is NULL!!");
+		goto ret_error;
+	}
+	if (WEBCLIENT_CONF_MAX_ENTITY_SIZE < strlen(request->entity)) {
+		PRNT("WEBCLIENT_CONF_MAX_ENTITY_SIZE < strlen(request->entity)");
+		goto ret_error;
+	}
+#ifdef CONFIG_NET_SECURITY_TLS
+	if (ssl_conf) {
+		request->tls = true;
+		memcpy(&request->ssl_config, ssl_conf, sizeof(struct http_client_ssl_config_t));
+
+		data = mallinfo();
+		if (data.fordblks < WEBCLIENT_CONF_MIN_TLS_MEMORY) {
+			PRNT("Error: Not enough memory!!");
+			goto ret_error;
+		}
+	}
+#endif
+
+	request->buffer = (char *)zalloc(request->buflen);
+	if (request->buffer == NULL) {
+		PRNT("Error: Fail to allocate buffer");
+		return -1;
+	}
+
+	res = ws_test_client_send_request(request);
+	if (res != 0) {
+		PRNT("ws_test_client_send_request() failed %d", res);
+	}
+
+	free(request->buffer);
+	FUNC_EX;
+	return res;
+
+ret_error:
+	FUNC_EX;
+	return -1;
+}
+
+static int ws_test_prepare_request(int argc, char** argv, struct http_client_request_t *request)
+{
+	FUNC_EN;
+	char *p, *q;
+	g_is_test_entity = 0;
+	g_keep_alive_count = 0;
+	g_chunk_count = 0;
+	g_trailer = 0;
+	memset(request, 0, sizeof(struct http_client_request_t));
+
+
+	if (!strncmp(argv[2], "GET", 4)) {
+		request->method = WGET_MODE_GET;
+	} else if (!strncmp(argv[2], "PUT", 4)) {
+		request->method = WGET_MODE_PUT;
+	} else if (!strncmp(argv[2], "POST", 5)) {
+		request->method = WGET_MODE_POST;
+	} else if (!strncmp(argv[2], "DELETE", 7)) {
+		request->method = WGET_MODE_DELETE;
+	} else {
+		PRNT("invalid req_type");
+		return -1;
+	}
+
+	request->url = argv[3];
+
+#ifdef CONFIG_NET_SECURITY_TLS
+	if (!strncmp(request->url, "https", 5)) {
```

Comment:
It is better to compare with 6.

Reply:
The URL provided by the user will be in this format -> https://127.0.0.1/
`webserver_test 1 GET https://127.0.0.1/`
So only 5 characters needs to be checked. The URL is further validated.